src/main/scala/edu/knowitall/srlie/SrlExtraction.scala

```diff
@@ -322,6 +357,8 @@ object SrlExtraction {
     def concat(other: Relation) = {
       Relation(this.text + " " + other.text, None, this.tokens ++ other.tokens, this.intervals ++ other.intervals)
     }
+
+    override def hashCode: Int = super.hashCode
```

Comment:
This is wrong. You need to ensure that equality implies equal hash codes; that is, if two objects are equal via `equals`, they will always hash to the same location. This looks to be delegating to `Object.equals`, which compares object identity, not value equality.